osbs/api.py

```diff
@@ -394,7 +396,25 @@ def _create_build_config_and_build(self, build_request):
                                                        scheduled=True)
             logger.debug('Changed parent ImageStreamTag? %s', changed_ist)
 
+        old_nodeselector = existing_bc['spec'].get('nodeSelector')
+        logger.debug('saving build config\'s nodeSelector %s', old_nodeselector)
+
+        if old_nodeselector:
+            existing_bc = self._get_existing_build_config(existing_bc)
+
+            existing_bc['spec']['nodeSelector'] = nodeselector
+            logger.debug('setting build config\'s nodeSelector %s', nodeselector)
+            self.os.update_build_config(build_config_name, json.dumps(existing_bc))
```

Comment:
Instead of 'fetch BC, update local json and update BC' we might want to use HTTP PATCH - https://docs.openshift.com/container-platform/3.4/rest_api/openshift_v1.html#partially-update-the-specified-buildconfig